fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -227,4 +158,151 @@ def shell_escaped_password(password)
       "'" + password + "'"
     end
   end
+
+  # Generates commands and executes the iTMSTransporter by invoking its Java app directly, to avoid the crazy parameter
+  # escaping problems in its accompanying shell script.
+  class JavaTransporterExecutor < TransporterExecutor
+    def build_upload_command(username, password, source = "/tmp")
+      [
+        Helper.transporter_java_executable_path.shellescape,
+        "-Djava.ext.dirs=#{Helper.transporter_java_ext_dir.shellescape}",
+        '-XX:NewSize=2m',
+        '-Xms32m',
+        '-Xmx1024m',
+        '-Xms1024m',
+        '-Djava.awt.headless=true',
+        '-Dsun.net.http.retryPost=false',
+        "-classpath #{Helper.transporter_java_jar_path.shellescape}",
+        'com.apple.transporter.Application',
+        '-m upload',
+        "-u #{username.shellescape}",
+        "-p #{password.shellescape}",
+        "-f #{source.shellescape}",
+        ENV["DELIVER_ITMSTRANSPORTER_ADDITIONAL_UPLOAD_PARAMETERS"], # that's here, because the user might overwrite the -t option
+        '-t Signiant',
+        '-k 100000',
+        '2>&1' # cause stderr to be written to stdout
+      ].compact.join(' ') # compact gets rid of the possibly nil ENV value
+    end
+
+    def build_download_command(username, password, apple_id, destination = "/tmp")
+      [
+        Helper.transporter_java_executable_path.shellescape,
+        "-Djava.ext.dirs=#{Helper.transporter_java_ext_dir.shellescape}",
+        '-XX:NewSize=2m',
+        '-Xms32m',
+        '-Xmx1024m',
+        '-Xms1024m',
+        '-Djava.awt.headless=true',
+        '-Dsun.net.http.retryPost=false',
+        "-classpath #{Helper.transporter_java_jar_path.shellescape}",
+        'com.apple.transporter.Application',
+        '-m lookupMetadata',
+        "-u #{username.shellescape}",
+        "-p #{password.shellescape}",
+        "-apple_id #{apple_id.shellescape}",
+        "-destination #{destination.shellescape}",
```

Comment:
@KrauseFx what do you think? I carried this over because it part of the original command.

I have to admit, though, I don't understand how it would "overwrite the -t option" as the comment describes.